src/reporters/json.ts

```diff
@@ -219,6 +250,81 @@ function stepInfo(
   };
 }
 
+async function processScreenshot(screenshot: Buffer) {
+  const img = sharp(screenshot, { sequentialRead: true });
+  const { width, height } = await img.metadata();
+  /**
+   * Chop the screenshot image (1280*720) which is the default
+   * viewport size in to 64 equal blocks for a given image
+   * which can be acheived by keeping the division to 8
+   *
+   * Changing division to 16 would yield us 256 smaller blocks, but it is not
+   * optimal for caching in the ES and proved out to be bad, so we stick to 8
+   */
+  const divisions = 8;
+  const blockWidth = width / divisions;
+  const blockHeight = height / divisions;
+  const reference: ScreenshotReference = {
+    width,
+    height,
+    blockWidth,
+    blockHeight,
+    blocks: [],
+  };
+  const blocks: Array<ScreenshotBlob> = [];
+
+  for (let row = 0; row < divisions; row++) {
+    const top = row * blockHeight;
+    for (let col = 0; col < divisions; col++) {
+      const left = col * blockWidth;
+      const buf = await img
+        .extract({ top, left, width: blockWidth, height: blockHeight })
+        .webp({ quality: 80 })
+        .toBuffer();
+
+      const hash = createHash('sha1').update(buf).digest('hex');
+      blocks.push({
+        blob: buf.toString('base64'),
+        id: hash,
+      });
+      /**
+       * We dont write the width, height of individual blocks on the
+       * reference as we use similar sized blocks for each extraction,
+       * we would need to send the width and height here if we decide to
+       * go with dynamic block extraction.
+       */
+      reference.blocks.push({
+        hash,
+        top,
```

Comment:
Since we dont have dynamic extraction at this point, we could save some more size by calculating the `top`, `left` using the number of blocks and fixed block width and height. 

num of blocks = divisions ^ 2
calculating top and left the same way as we do here for extraction, we do the same for composition on the UI. 

Thanks to @axw for the idea.

Reply:
Compared to the size the images storing the dimensions and coordinates is in significant, especially if we don't index them which we don't have to. It's more important I think to have a future proof format here then to save on a few fields. That way we do want a more sophisticated way of slicing up images the client will not have to change. I left a similar note in the type definition elsewhere

Reply:
I thought about a bit around future proofing the code on the client side. If we are not going to use `canvas` on the Kibana UI for composing and can use `sharp` or image processing library, we don't even need to send these information except for the hash.

Reply:
I had changed as per your comment, but lets come back to it based on our solution on the UI side.

Reply:
Yeah, this isn't implementation specific, but rather I'd like us to support non-grid layouts in the future. Rather than it being a list of blocks of a hard coded size let's think of it as a list of compositing operations. It's far more powerful, isn't extra work, and gives us more flexibility